projects/core/src/checkout/http-interceptors/checkout-cart.interceptor.spec.ts

```diff
@@ -0,0 +1,206 @@
+    routingService = TestBed.inject(RoutingService);
+    multiCartService = TestBed.inject(MultiCartService);
+
+    spyOn(routingService, 'go');
+    spyOn(multiCartService, 'reloadCart');
+  });
+
+  describe('intercept', () => {
+    it('should redirect to cart page if the cart is not found', () => {
+      spyOn(routingService, 'getRouterState').and.returnValue(
+        of({ state: { semanticRoute: 'checkoutDelivery' } } as any)
+      );
+
+      http
+        .get('/test')
+        .pipe(catchError((error: any) => throwError(error)))
+        .subscribe(
+          (_result) => {},
+          (error) => ({
+            if(this) {
+              this.error = error;
+            },
+          })
+        );
+
+      const mockReq = httpMock.expectOne((req) => {
+        return req.method === 'GET';
+      });
+      mockReq.flush(
+        {
+          errors: [
+            {
+              type: 'CartError',
+              message: 'Cart not found',
+              reason: 'notFound',
+              subject: '123',
+            },
+          ],
+        },
+        { status: 400, statusText: 'Error' }
+      );
+
+      expect(routingService.go).toHaveBeenCalledWith({ cxRoute: 'cart' });
+      expect(multiCartService.reloadCart).toHaveBeenCalledWith('123');
+    });
+
+    it('should NOT do anything if the route is not part of checkout', () => {
+      spyOn(routingService, 'getRouterState').and.returnValue(
+        of({ state: { semanticRoute: 'orderConfirmation' } } as any)
+      );
+
+      http
+        .get('/test')
+        .pipe(catchError((error: any) => throwError(error)))
+        .subscribe(
+          (_result) => {},
+          (error) => ({
+            if(this) {
+              this.error = error;
+            },
+          })
+        );
+
+      const mockReq = httpMock.expectOne((req) => {
+        return req.method === 'GET';
+      });
+      mockReq.flush(
+        {
+          errors: [
+            {
+              type: 'CartError',
+              message: 'Cart not found',
+              reason: 'notFound',
+              subject: '123',
+            },
+          ],
+        },
+        { status: 400, statusText: 'Error' }
+      );
+
+      expect(routingService.go).not.toHaveBeenCalled();
+      expect(multiCartService.reloadCart).not.toHaveBeenCalled();
+    });
+
+    it('should NOT reload cart if there is no subject', () => {
+      spyOn(routingService, 'getRouterState').and.returnValue(
+        of({ state: { semanticRoute: 'checkoutDelivery' } } as any)
+      );
+
+      http
+        .get('/test')
+        .pipe(catchError((error: any) => throwError(error)))
+        .subscribe(
+          (_result) => {},
+          (error) => ({
+            if(this) {
+              this.error = error;
+            },
+          })
+        );
+
+      const mockReq = httpMock.expectOne((req) => {
+        return req.method === 'GET';
+      });
+      mockReq.flush(
+        {
+          errors: [
+            {
```

Comment:
This one can be extracted to a constant or generated by a function to avoid duplication